packages/token-balance-cache/src/providers/erc721.ts

```diff
@@ -0,0 +1,68 @@
+import Web3 from 'web3';
+import { BigNumber, providers } from 'ethers';
+import type { ERC721 } from 'common-common/src/eth/types';
+import { ERC721__factory } from 'common-common/src/eth/types';
+
+import type { IChainNode } from '../types';
+import { BalanceProvider } from '../types';
+import { BalanceType } from 'common-common/src/types';
+
+type EthBPOpts = {
+  tokenAddress?: string;
+  contractType?: string;
+};
+
+export default class Erc721BalanceProvider extends BalanceProvider<EthBPOpts> {
+  public name = 'erc721';
+  // Added Token Id as String because it can be a BigNumber (uint256 on solidity)
+  public opts = {
+    contractType: 'string?',
+    tokenAddress: 'string?',
+  };
+  public validBases = [BalanceType.Ethereum];
+
+  public getCacheKey(
+    node: IChainNode,
+    address: string,
+    opts: EthBPOpts
+  ): string {
+    return `${node.id}-${address}-${opts.tokenAddress || 'native'}`;
```

Comment:
we don't need "native" here! (or in the other token specific BPs)

Reply:
Makes sense, I forgot to read this function!